fecgov/openFEC
# Add org_tp and cmte_dsgn columns and related indexes to fec_fitem_sch…

## Summary (required)

- Resolves #3787 

Add org_tp and cmte_dsgn columns to Schedule A and B tables, backfill data for existing rows, and add indexes

## How to test the changes locally

Download the branch to your local machine.  Run migration file by either run 
```flyway migrate``` 
or 
```invoke create_sample_db```

Make sure migration run successfully.  
Connect to local DB, make sure 
1. these two columns are added to fec_fitem_sched_a_xxxx and fec_fitem_sched_b_xxxx tables.  
2. 4 indexes per partition tables of fec_fitem_sched_a_xxxx and fec_fitem_sched_b_xxxx are added for these 2 added columns.

select tablename, count(*)
from pg_indexes 
where tablename like 'fec_fitem_sched_a%'
and (indexname like 'idx_sched_a_%_org_tp%' or indexname like 'idx_sched_a_%_cmte_dsgn%')
group by tablename
order by tablename;

select tablename, count(*)
from pg_indexes 
where tablename like 'fec_fitem_sched_b%'
and (indexname like 'idx_sched_b_%_org_tp%' or indexname like 'idx_sched_b_%_cmte_dsgn%')
group by tablename
order by tablename;

select tablename, indexname
from pg_indexes 
where tablename like 'fec_fitem_sched_a%'
and (indexname like 'idx_sched_a_%_org_tp%' or indexname like 'idx_sched_a_%_cmte_dsgn%')
order by tablename, indexname;

select tablename, indexname
from pg_indexes 
where tablename like 'fec_fitem_sched_b%'
and (indexname like 'idx_sched_b_%_org_tp%' or indexname like 'idx_sched_b_%_cmte_dsgn%')
order by tablename, indexname;

Also, since the actual columns adding and indexes creating work are done, the same queries above run on DEV/STG/PRD should results the same results.

/* Optional:
To mimic the effect on DEV/STG/PRD, which already has the columns added and indexes created.
On your local machine, _AFTER_ migration of V150 finished, log in the local database, issue the following statement.  
```delete from flyway_schema_history where version = '0150';```  

then run ```flyway migrate``` again.  
It should run successfully without errors caused by objects already existed.
*/

The data had been backfilled on DEV/STG/PRD.  So queries cmte_dsgn/org_tp on DEV/STG/PRD can be checked as well.

The tsvector columns on DEV/STG/PRD in fec_fitem_sched_a_xxxx and fec_fitem_sched_b_xxxx had been updated with new definition defined in issue #3798.  One can check on the the correctness of these columns as well.


## Impacted areas of the application
List general components of the application that this PR will affect:

-  No API impact for this PR



## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()
